lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -335,7 +335,7 @@ class ClassMetadataInfo implements ClassMetadata
      * )
      * </pre>
      *
-     * @var array
+     * @psalm-var array<string, array<string, mixed>>
```

Comment:
```suggestion
     * @psalm-var array<string, array{name: string, query: string, isSelfClass: bool, resultSetMapping?: , resultClass?: class-string}>
```

as said in the above comment and also
inferred from `\Doctrine\ORM\Mapping\ClassMetadataFactory::addInheritedNamedNativeQueries` also I was not 100% sure about the `resultSetMapping` key of array 

the http://doctrine-project.org/schemas/orm/doctrine-mapping says that 
```xml
    <xs:attribute name="result-set-mapping" type="xs:string" />
```
the type is `string`
but if you look to this answer on stack overflow https://stackoverflow.com/a/37715649
it has different type - `array`